tests/app/views/test_suppliers.py

```diff
@@ -1356,6 +1371,26 @@ def test_adding_that_agreement_has_been_returned(self):
             assert_equal(data['frameworkInterest']['frameworkSlug'], 'digital-outcomes-and-specialists')
             assert_equal(data['frameworkInterest']['agreementReturned'], True)
             assert_equal(data['frameworkInterest']['agreementReturnedAt'], "2012-12-12T00:00:00.000000Z")
+            assert_equal(data['frameworkInterest']['countersigned'], False)
+            assert_is(data['frameworkInterest']['countersignedAt'], None)
+            assert_is(data['frameworkInterest']['signerDetails'], None)
+
+    def test_adding_that_agreement_has_been_countersigned(self):
+        with freeze_time('2012-12-12'):
+            response = self.supplier_framework_update(
+                0,
+                'digital-outcomes-and-specialists',
+                update={'countersigned': True}
+            )
+            assert_equal(response.status_code, 200)
+            data = json.loads(response.get_data())
+            assert_equal(data['frameworkInterest']['supplierId'], 0)
+            assert_equal(data['frameworkInterest']['frameworkSlug'], 'digital-outcomes-and-specialists')
+            assert_equal(data['frameworkInterest']['agreementReturned'], False)
+            assert_is(data['frameworkInterest']['agreementReturnedAt'], None)
+            assert_equal(data['frameworkInterest']['countersigned'], True)
```

Comment:
In real life, we wouldn't expect this to happen _**but**_ it seems fair enough to test this functionality as long as our API allows it.